internal/gengapic/genrest.go

```diff
@@ -245,36 +311,39 @@ func (g *generator) generateQueryString(m *descriptor.MethodDescriptorProto) {
 	}
 	sort.Strings(fields)
 
-	p("queryParamStrs := []string{}")
-	// TODO(dovs): make this work for nested fields
-	for _, f := range fields {
-		field := queryParams[f]
-		f = snakeToCamel(f)
+	requestObject := "req"
+
+	p("params := url.Values{}")
+	for _, path := range fields {
+		field := queryParams[path]
+		accessor := buildAccessor(path, false)
 		if field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_REPEATED {
 			// It's a slice, so check for nil
-			p("if req.Get%s() != nil {", f)
+			p("if %s%s != nil {", requestObject, accessor)
+		} else if field.GetProto3Optional() {
+			// Split right before the raw access
+			elts := strings.Split(path, ".")
+			elts = elts[:len(elts)-1]
+			p("if %[1]s%[2]s != nil && %[1]s%[3]s != nil {", requestObject, buildAccessor(strings.Join(elts, "."), false), buildAccessor(path, true))
```

Comment:
Could you put the results of the `buildAccessor` methods into local variables? This line is rather complex to grok at first glance what with the format specifier indices and nested function calls.

Reply:
Sure.